releasenotes-builder/src/test/java/com/github/checkstyle/MainTest.java

```diff
@@ -0,0 +1,206 @@
+// checkstyle: Checks Java source code and other text files for adherence to a set of rules.
+// Copyright (C) 2001-2022 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.github.checkstyle;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.kohsuke.github.GHIssue;
+import org.kohsuke.github.GHIssueState;
+import org.kohsuke.github.GHRepository;
+import org.mockito.MockedStatic;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import com.github.checkstyle.git.CsGit;
+import com.github.checkstyle.github.CsGitHub;
+import com.github.checkstyle.globals.Constants;
+import com.github.checkstyle.internal.GhIssueUtil;
+import com.github.checkstyle.internal.RevCommitUtil;
+import com.github.checkstyle.publishers.MailingListPublisher;
+import com.github.checkstyle.publishers.SourceforgeRssPublisher;
+import com.github.checkstyle.publishers.TwitterPublisher;
+
+public class MainTest {
+    private static final Set<RevCommit> TEST_COMMITS = new HashSet<>();
+
+    private static final Set<GHIssue> TEST_ISSUES = new HashSet<>();
+
+    private static final Answer<GHIssue> ISSUE_ANSWER = new Answer<GHIssue>() {
+        @Override
+        public GHIssue answer(InvocationOnMock invocation) throws Throwable {
+            final int findIssue = invocation.getArgument(0);
+            GHIssue result = null;
+
+            for (GHIssue item : TEST_ISSUES) {
+                if (item.getNumber() == findIssue) {
+                    result = item;
+                    break;
+                }
+            }
+
+            return result;

```

Comment:
I've confirmed the real GH will not return a `null`. It will throw an exception if the issue doesn't exist.

Example:
````
        final GHRepository remoteRepo = CsGitHub.createRemoteRepo(null, "checkstyle/checkstyle");

        final GHIssue issue = remoteRepo.getIssue(999999);

        System.out.println(issue.toString());
````
Output:
````
Exception in thread "main" org.kohsuke.github.GHFileNotFoundException: https://api.github.com/repos/checkstyle/checkstyle/issues/999999 {"message":"Not Found","documentation_url":"https://docs.github.com/rest/reference/issues#get-an-issue"}
	at org.kohsuke.github.GitHubClient.interpretApiError(GitHubClient.java:588)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:449)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:403)
	at org.kohsuke.github.Requester.fetch(Requester.java:85)
	at org.kohsuke.github.GHRepository.getIssue(GHRepository.java:409)
	at com.github.checkstyle.github.Run.main(Run.java:13)
Caused by: java.io.FileNotFoundException: https://api.github.com/repos/checkstyle/checkstyle/issues/999999
	at org.kohsuke.github.GitHubConnectorResponseErrorHandler$1.onError(GitHubConnectorResponseErrorHandler.java:60)
	at org.kohsuke.github.GitHubClient.detectKnownErrors(GitHubClient.java:472)
	at org.kohsuke.github.GitHubClient.sendRequest(GitHubClient.java:434)
	... 4 more
````

I will incorporate and probably start an issue for this.